.github/workflows/e2e_docker.yml

```diff
@@ -0,0 +1,65 @@
+name: E2E (docker)
+
+on:
+  push:
+  pull_request:
+    branches: [main]
+
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label ||
+    github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+jobs:
+  e2e:
+    if: github.ref == 'refs/heads/main' || github.event_name == 'pull_request'
```

Comment:
what events is this condition trying to avoid?

Reply:
this condition combined with the on condition:
```
on:
  push:
  pull_request:
    branches: [main]
```
ensures that the tests only run on a pull_request made to `main` branch or when  a branch commit is pushed in to the `main` branch

Reply:
I see that. I'm asking what conditions you're trying to avoid, because that sounds like almost all of them.

Reply:
well it specifically avoids these conditions:
- commits pushed to branches other than `main` i.e all branches regardless of whether they have a pull request open or not. this includes random testing branches that anyone can create
- pull requests with branches other than `main` as base since they are not as important to test. 
- sometimes a scenario can also occur where a branch can have a pull request open and any commit that is pushed to it triggers the workflows twice: once for the `push` condition and once for the `pull_request` condition

Reply:
can't all of those be excluded by not selecting them in the `on` expression?

Reply:
I suppose it can be done. Thanks for the suggestion. I've changed the `on` condition accordingly